### DISCUSSION TITLE
RTL Support: postcss-rtl directives

### DISCUSSION THREAD

**Discussion by bmoellers** [vercel/next.js#13836] [Help]
So I hooked up postcss-rtl to add rtl support to our next JS project. It was pretty easy to do and the following postcss.config.js file works great:

```
module.exports = {
  plugins: {
    'postcss-flexbugs-fixes': {},
    'postcss-preset-env': {
      autoprefixer: {
        flexbox: 'no-2009',
      },
      stage: 3,
      features: {
        'custom-properties': false,
      },
    },
    'postcss-rtl': {},
  },
};
```

The issue I am having is that postcss-rtl relies on comments to provide directives that allow you control the way things happen. Here is an example where postcss-rtl is changing things on a custom drawn check mark and just trying to get the transform function to work. If the directive works the CSS would work for RTL:

```
      &::after {
        position: absolute;
        top: 5px;
        left: 3px;
        content: none;
        display: flex;
        height: 4px;
        width: 9px;
        border-left: 2px solid;
        border-bottom: 2px solid;
        border-color: $ps-blue;
        transform: rotate(-45deg) /* rtl: rotate(-45deg) scaleX(-1) */;
      }
```

What I am seeing is that comments are being stripped somewhere in the process. I tried removing the other two postcss plugins and rearranging order but it still seems that the comments are removed.

Is there a way in next js to not strip comments in CSS until the end of the postcss chain (note postcss-rtl by default removes the comments in their step)?

**Comment by AkpolatKadir** [comment]
Hi, Any updates on this issue? I am facing the same and I can't find a solution.

**Comment by bab2683** [comment]
> So I hooked up postcss-rtl to add rtl support to our next JS project. It was pretty easy to do and the following postcss.config.js file works great:
> 
> ```
> module.exports = {
>   plugins: {
>     'postcss-flexbugs-fixes': {},
>     'postcss-preset-env': {
>       autoprefixer: {
>         flexbox: 'no-2009',
>       },
>       stage: 3,
>       features: {
>         'custom-properties': false,
>       },
>     },
>     'postcss-rtl': {},
>   },
> };
> ```
> 
> The issue I am having is that postcss-rtl relies on comments to provide directives that allow you control the way things happen. Here is an example where postcss-rtl is changing things on a custom drawn check mark and just trying to get the transform function to work. If the directive works the CSS would work for RTL:
> 
> ```
>       &::after {
>         position: absolute;
>         top: 5px;
>         left: 3px;
>         content: none;
>         display: flex;
>         height: 4px;
>         width: 9px;
>         border-left: 2px solid;
>         border-bottom: 2px solid;
>         border-color: $ps-blue;
>         transform: rotate(-45deg) /* rtl: rotate(-45deg) scaleX(-1) */;
>       }
> ```
> 
> What I am seeing is that comments are being stripped somewhere in the process. I tried removing the other two postcss plugins and rearranging order but it still seems that the comments are removed.
> 
> Is there a way in next js to not strip comments in CSS until the end of the postcss chain (note postcss-rtl by default removes the comments in their step)?

Separate the rule and apply the directive to that part, directives in postcss-rtl work on the whole declaration, not individual properties, look [here](https://www.npmjs.com/package/postcss-rtl#ignoring-specific-declarations).

**Reply by SarkarKurdish** [reply]
This does not work for me i'm using scss which i don't think makes any difference

this is my postcss file

`module.exports = {
  plugins: {
    tailwindcss: {},
    "postcss-flexbugs-fixes": {},
    "postcss-preset-env": {
      autoprefixer: {
        flexbox: "no-2009",
      },
      stage: 3,
      features: {
        "custom-properties": false,
      },
    },
    "postcss-rtlcss": {},
  },
};
`

**Comment by YahyaAhmad** [comment]
Is there any update here?
